### PR TITLE
Auto-completion feature for bash

### PR DESCRIPTION
**Pull request by mohitk108**
### Guide for adding auto-completion:

- Run the command 
`./gradlew installDist`
- The above command will create script in the build/install directory, and we can use them to run the application like a command line tool.
- Source the script correctly in the .bashrc file by adding the following line
`source /your/path/to/kotlin-cli-completion.sh`
- In my case for example, the line looks like:
`source /home/claddy/Documents/Projects/MempoolCLI/kotlin-cli-completion.sh`
- Now, set an Alias, by adding the following line
`alias kotlin-cli="/path/to/your/build/install/mac/bin/your-application-name" `
- On my case for example, the line looks like,
`alias kotlin-cli="/home/claddy/Documents/Projects/MempoolCLI/build/install/mac/bin/mac" `
- Finally, source the file 
`source ~/.bashrc`
- Now run the command 
 `kotlin-cli fetchBlockId -s 730000`
- Use `Tab` button to play around the auto completion.

### CODE REVIEW

File: kotlin-cli-completion.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# kotlin-cli-completion.sh
```
Comment (t-bast) [inline kotlin-cli-completion.sh]: That kind of files are usually placed in a `contrib` folder, not directly at the top-level of the project.
Also, it should be named `mempool-cli.bash-completion`.
Please add comments in this file linking the tutorials/instructions you used to create this file.

**Review by t-bast — COMMENTED** [review]
I have a few issues with the current state of the project, that need to be resolved.

First of all, some tests are failing: this should never happen, you shouldn't have merged a pull request to `master` with failing tests, it shows that you're not spending enough time testing your work. It is very important that you only ask reviews once you've spent enough time making sure that everything is working.

Second, the instructions to run the CLI in `README.md` are not working: if I run `./gradlew build` and then `java -jar build/libs/mac-1.0-SNAPSHOT.jar` it returns an error `no main manifest attribute, in build/libs/mac-1.0-SNAPSHOT.jar`. This makes me think that you have not tested the auto-complete yourself?

Last, the instructions that you put in this PR's description to explain how to use the auto-complete must be added to the README, otherwise how will users be able to make it work?

**Comment by mohitk108** [comment]
Issues fixed recently:
- Failing tests in 04028374f111fcdbf23a08d9b9f71135f30fb874
- Instruction `java -jar build/libs/mac-1.0-SNAPSHOT.jar` returning error `no main manifest attribute, in build/libs/mac-1.0-SNAPSHOT.jar` in 175ba4bd8d1f908746f81aea440ca7ea46202114
- Readme is updated using multiple commits 91d72704f479a7c079c13e30e419db2d76ba0d49 & 9b22ffaf0e3a7b93037be3878994979123ac4b77
- Changes related to mempool-cli.bash-completion in 0b97e4761d3ea07e07eb53ac31783dcd52f41c63 
- Tutorial references added in 1b62a593933b7fd918e9513fa1ac32f6c331bb22

**Comment by mohitk108** [comment]
- I'd planned to fix the failing tests once the part of auto-completion is completed, and reviewed by you.
- I did test out the autocompletion, but I never used the java -jar command to do it. Instead I preferred ./gradlew run command.
- I souced the mempool-cli command directly into my terminal, so even when in the root directory, I accessed it using `mempool-cli fetchBlockId -s 730000`.

And I'm really sorry to irritate you with frequent code-review request. 
I sincerely apologise.

**Comment by t-bast** [comment]
> And I'm really sorry to irritate you with frequent code-review request.
> I sincerely apologise.

Don't get me wrong, I'm not irritated and you have nothing to apologize for, this is expected during the learning process!

I am making many comments on things that must be improved, but that should never be taken personally, those are comments on code and work results/process. The goal is to help you learn important things that are needed to successfully work on open source projects.

The most important thing for open source contributions is detailed communication. For example, I'm discovering while testing this PR that some tests are failing and build instructions are not working. If that's something you were already aware of, you should detail that in the PR description to let me know that I should ignore it (or that I should ask you to fix it). Reviewers cannot know what you think if you don't write it in the PR description or in comments: it is thus very important to write detailed PR descriptions to give reviewers all the information they need. For this specific PR, you should also have detailed what you tested and how you tested it: otherwise as a reviewer that sees that build instructions don't work, I cannot know if you even tested it.

Then the second most important thing is to accept that reviewers have a lot of work, and won't be available immediately to review your code or answer your questions. You should make sure you've already reviewed and tested your code yourself, and it's the best result you could get before asking other people for review. Once you've asked for reviews, you should wait for reviewers to provide it whenever they can: if you ping reviewers aggressively, they will on the contrary ignore you and focus on other things that are more important to them. I'm doing my best to review your work as soon as possible.

A last important point is to be independent: while you're waiting for someone to review some of your work, you should always have other things to do or explore in parallel. You should never stop working because you're waiting for a review, otherwise you'll be wasting a lot of precious time. There are always things to do, things to improve, things to learn.

None of this feedback is negative or against you: it's just important advice to make you more successful!

**Review by t-bast — APPROVED** [review]

**Pull request merged**